src/torchmetrics/detection/iou.py

```diff
@@ -0,0 +1,169 @@
+                [num_boxes] containing detection scores for the boxes.
+            - ``labels``: ``torch.IntTensor`` of shape
+                [num_boxes] containing 0-indexed detection classes for the boxes.
+            target: A list consisting of dictionaries each containing the key-values
+            (each dictionary corresponds to a single image):
+            - ``boxes``: ``torch.FloatTensor`` of shape
+                [num_boxes, 4] containing `num_boxes` ground truth boxes of the format
+                specified in the contructor. By default, this method expects
+                [xmin, ymin, xmax, ymax] in absolute image coordinates.
+            - ``labels``: ``torch.IntTensor`` of shape
+                [num_boxes] containing 1-indexed ground truth classes for the boxes.
+        Raises:
+            ValueError:
+                If ``preds`` is not of type List[Dict[str, Tensor]]
+            ValueError:
+                If ``target`` is not of type List[Dict[str, Tensor]]
+            ValueError:
+                If ``preds`` and ``target`` are not of the same length
+            ValueError:
+                If any of ``preds.boxes``, ``preds.scores``
+                and ``preds.labels`` are not of the same length
+            ValueError:
+                If any of ``target.boxes`` and ``target.labels`` are not of the same length
+            ValueError:
+                If any box is not type float and of length 4
+            ValueError:
+                If any class is not type int and of length 1
+            ValueError:
+                If any score is not type float and of length 1
+        """
+        _input_validator(preds, target)
+
+        for item in preds:
+            det_boxes = _fix_empty_tensors(item["boxes"])
+            det_boxes = box_convert(det_boxes, in_fmt=self.box_format, out_fmt="xyxy")
+            self.detections.append(det_boxes)
+            self.detection_labels.append(item["labels"])
+            self.detection_scores.append(item["scores"])
+
+        for item in target:
+            gt_boxes = _fix_empty_tensors(item["boxes"])
+            gt_boxes = box_convert(gt_boxes, in_fmt=self.box_format, out_fmt="xyxy")
+            self.groundtruths.append(gt_boxes)
+            self.groundtruth_labels.append(item["labels"])
+
+    def _get_classes(self) -> List:
+        """Returns a list of unique classes found in ground truth and detection data."""
+        if len(self.detection_labels) > 0 or len(self.groundtruth_labels) > 0:
+            return torch.cat(self.detection_labels + self.groundtruth_labels).unique().tolist()
+        return []
+
+    def compute(self) -> dict:
+        """Computes IoU based on inputs passed in to ``update`` previously."""
+        preds = torch.cat(self.detections)
+        target = torch.cat(self.groundtruths)
+        labels = torch.cat(self.detection_labels)
+        result = self.iou_fn.__func__(preds, target, self.iou_threshold)
```

Comment:
stupid question: why is the ``__func__`` necessary?

Reply:
Because I assigned a static function to an instance variable for inheritance and it would otherwise pass the self var as an argument as well